packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/GeoSpatialTests.kt

```diff
@@ -0,0 +1,229 @@
+package io.realm.kotlin.test.mongodb.common
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.annotations.ExperimentalGeoSpatialApi
+import io.realm.kotlin.entities.Location
+import io.realm.kotlin.entities.sync.SyncRestaurant
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.internal.platform.runBlocking
+import io.realm.kotlin.log.LogLevel
+import io.realm.kotlin.mongodb.User
+import io.realm.kotlin.mongodb.subscriptions
+import io.realm.kotlin.mongodb.sync.SyncConfiguration
+import io.realm.kotlin.mongodb.syncSession
+import io.realm.kotlin.test.mongodb.TEST_APP_FLEX
+import io.realm.kotlin.test.mongodb.TestApp
+import io.realm.kotlin.test.mongodb.createUserAndLogIn
+import io.realm.kotlin.test.util.TestHelper
+import io.realm.kotlin.test.util.use
+import io.realm.kotlin.types.geo.Distance
+import io.realm.kotlin.types.geo.GeoBox
+import io.realm.kotlin.types.geo.GeoCircle
+import io.realm.kotlin.types.geo.GeoPoint
+import io.realm.kotlin.types.geo.GeoPolygon
+import kotlinx.coroutines.delay
+import org.mongodb.kbson.ObjectId
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFails
+import kotlin.time.Duration.Companion.seconds
+
+private val GEO_SCHEMA = setOf(SyncRestaurant::class, Location::class)
+
+@OptIn(ExperimentalGeoSpatialApi::class)
+class GeoSpatialTests {
+    private lateinit var app: TestApp
+
+    @BeforeTest
+    fun setup() {
+        app = TestApp(this::class.simpleName, appName = TEST_APP_FLEX, logLevel = LogLevel.ALL)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    private fun createRandomUser(): User =
+        runBlocking {
+            app.createUserAndLogIn(
+                email = TestHelper.randomEmail(),
+                password = "password1234"
+            )
+        }
+
+    @Test
+    fun write() {
+        val user = createRandomUser()
+
+        val config =
+            SyncConfiguration.Builder(
+                user = user,
+                schema = GEO_SCHEMA
+            ).initialSubscriptions(rerunOnOpen = true) {
+                add(it.query<SyncRestaurant>())
+            }.build()
+
+        Realm.open(config).use {
+            it.writeBlocking {
+                copyToRealm(SyncRestaurant())
+            }
+        }
+    }
+
+    @Test
+    fun write_outsideSubscriptionsFail() {
+        val user = createRandomUser()
+
+        val config =
+            SyncConfiguration.Builder(
+                user = user,
+                schema = GEO_SCHEMA
+            ).initialSubscriptions(rerunOnOpen = true) {
+            }.build()
+
+        Realm.open(config).use {
+            it.writeBlocking {
+                assertFails {
```

Comment:
I don't think the write will fail as long as the class is in the schema, instead you will probably see a compensating write?

Reply:
It fails with `RLM_ERR_NO_SUBSCRIPTION_FOR_WRITE`. It is core preemptively catching the error.